internal/envoygateway/config/runner/runner.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package runner
+
+import (
+	"context"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/logging"
+)
+
+const runnerName = "config-watcher"
+
+type Runner struct {
+	cfgPath string
+	cfg     *config.Server
+	logger  logging.Logger
+}
+
+func New(cfgPath string, cfg *config.Server) *Runner {
+	return &Runner{
+		cfgPath: cfgPath,
+		cfg:     cfg,
+		logger:  cfg.Logger.WithName(runnerName),
+	}
+}
+
+func (r *Runner) Name() string {
+	return runnerName
+}
+
+func (r *Runner) Start(ctx context.Context) error {
+	if r.cfgPath == "" {
+		r.logger.Info("no config file provided, skipping config watcher")
+		return nil
+	}
+
+	watcher, err := fsnotify.NewWatcher()
```

Comment:
Is possible to reuse the notifier under `internal/provider/file/notifier.go` ?

Reply:
still WIP, will try it later.

Reply:
IMO `internal/provider/file/notifier.go` had not user-firendly API, I will try to use lib from hugo in this PR, cause it's not the key point.